From 39937687264e071e1b846e9ae38b10568e8b55fc Mon Sep 17 00:00:00 2001
From: Rajnikant Bhojani <rajnikant.bhojani@xilinx.com>
Date: Thu, 23 Jan 2020 19:50:34 +0530
Subject: [PATCH 0984/1852] usb: gadget: f_tcm: add TARGET_SCF_ACK_KREF flag in
 BOT

commit 281fab67ae232c86d2e264db4eb465daffed9366 from
https://github.com/Xilinx/linux-xlnx.git

in BOT mode se_cmd and I/O resource getting free during target
fabric module operation check_stop_free which does not gauranteed
freeing of resource after actual usb bus transfer. adding
TARGET_SCF_ACK_KREF flag during submitting target request will
increment additinal cmd_kref which will prevent deallocating
se_cmd and I/O resource before actual usb bus transfer.

Signed-off-by: Rajnikant Bhojani <rajnikant.bhojani@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Piyush Mehta <piyush.mehta@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/usb/gadget/function/f_tcm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/function/f_tcm.c b/drivers/usb/gadget/function/f_tcm.c
index fd8495ed6f07..19fb570fd0aa 100644
--- a/drivers/usb/gadget/function/f_tcm.c
+++ b/drivers/usb/gadget/function/f_tcm.c
@@ -1287,7 +1287,8 @@ static void bot_cmd_work(struct work_struct *work)
 
 	if (target_submit_cmd(se_cmd, tv_nexus->tvn_se_sess,
 			cmd->cmd_buf, cmd->sense_iu.sense, cmd->unpacked_lun,
-			cmd->data_len, cmd->prio_attr, dir, 0) < 0)
+			cmd->data_len, cmd->prio_attr, dir,
+			TARGET_SCF_ACK_KREF) < 0)
 		goto out;
 
 	return;
-- 
2.31.1

